lib/src/actions/package/providers/xbps.rs

```diff
@@ -0,0 +1,115 @@
+use super::PackageProvider;
+use crate::actions::package::repository::PackageRepository;
+use crate::actions::package::PackageVariant;
+use crate::atoms::command::Exec;
+use crate::steps::Step;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::process::Command;
+use tracing::{debug, trace, warn};
+use which::which;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Xbps {}
+
+impl PackageProvider for Xbps {
+    fn name(&self) -> &str {
+        "Xbps"
+    }
+
+    fn available(&self) -> bool {
+        let mut install = false;
+        let mut query = false;
+        match which("xbps-install") {
+            Ok(_) => {
+                install = true;
+            }
+            Err(_) => {
+                warn!(message = "xbps-install not available");
+            }
+        };
+        match which("xbps-query") {
+            Ok(_) => {
+                query = true;
+            }
+            Err(_) => {
+                warn!(message = "xbps-query not available");
+            }
+        };
+        install && query
```

Comment:
Can the query command be installed in bootstrap if missing?

Reply:
No, void ships with both these binaries as inside single package called  xbps, so if one is installed other would also be installed